From 6ecb671d2c24772146b8ecdc452cfb9a10714adc Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Tue, 5 Nov 2024 13:25:12 +0100
Subject: [PATCH 1420/1695] module: scmi_clock: handle SCMI_DENIED in
 set_request_respond()

Handle FWK_E_ACCESS to SCMI_DENIED conversion in set_request_respond()
to provide more information about errors and avoid sending a generic
error.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0aae86e01664ed0819dfefe439d04cd92425e3d7 module: scmi_clock: handle SCMI_DENIED in set_request_respond()]
Change-Id: I8a74a2931c992a37063b139bb53e6bea52f49688
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/416481
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c         | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c b/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
index 01f2627cc..e6beaf8ef 100644
--- a/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
+++ b/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
@@ -616,6 +616,8 @@ static void set_request_respond(fwk_id_t service_id, int status)
         return_values.status = (int32_t)SCMI_INVALID_PARAMETERS;
     } else if (status == FWK_E_SUPPORT) {
         return_values.status = (int32_t)SCMI_NOT_SUPPORTED;
+    } else if (status == FWK_E_ACCESS) {
+        return_values.status = (int32_t)SCMI_DENIED;
     } else if (status != FWK_SUCCESS) {
         return_values.status = (int32_t)SCMI_GENERIC_ERROR;
     } else {
-- 
2.47.2

